source/develop/release-management/features/network/allowExplicitDnsConfiguration.html.md

```diff
@@ -0,0 +1,196 @@
+configuration in resolv.conf. If we now setup configuration in
+NetworkAttachment, attaching ManagementNetwork of given DataCenter (or
+later of given Cluster) to specific Host, this configuration is more
+specific than one in Network of DC (or later Cluster) scope and will
+be used instead.  
+
+### DB
+Database needs to be updated so it can accommodate DNS configuration.
+Two places will be altered:
+
+* `network_attachments` table
+* network table
+
+both these tables must accommodate multiple DNS records. Naive
+solution would be to store them comma separated into single column.
+Better solution would be creating separate table for that. That's
+subject of further discussion. VDSM supports 0 to 3 DNS entries, so
+engine needs to comply to this limitation as well.
+
+### REST
+
+As mentioned, you can specify DNS Configuration at three places.
+Corresponding REST areas will be altered. But first we need to add new
+element 'dns_configuration' as:
+ 
+```
+@Type
+public interface DnsResolverConfiguration {
+ String[] nameServers();
+}
+```
+
+…, which will be translated into following XSchema definition during
+engine build.
+
+
+```
+<xs:element name="dns_resolver_configuration" type="DnsResolverConfiguration"/>
+
+  <xs:complexType name="DnsResolverConfiguration">
+    <xs:sequence>
+      <xs:element maxOccurs="1" minOccurs="0" name="name_servers">
+        <xs:complexType>
+          <xs:annotation>
+            <xs:appinfo>
+              <jaxb:class name="NameServersList"/>
+            </xs:appinfo>
+          </xs:annotation>
+          <xs:sequence>
+            <xs:element maxOccurs="unbounded" minOccurs="0" name="name_server" type="xs:string">
+              <xs:annotation>
+                <xs:appinfo>
+                  <jaxb:property name="NameServers"/>
+                </xs:appinfo>
+              </xs:annotation>
+            </xs:element>
+          </xs:sequence>
+        </xs:complexType>
+      </xs:element>
+    </xs:sequence>
+  </xs:complexType>
+```
+
+After doing so, we're prepared to use following xml fragment defining
+DNS configuration in several places (read on):
+
+```
+<dns_resolver_configuration>
+  <name_servers>
+    <name_server>1.1.1.1</name_server>
+    <name_server>2.2.2.2</name_server>
+  </name_servers>
+</dns_resolver_configuration>
+```
+
+#### Creating new Host
+```
+POST /ovirt-engine/api/hosts HTTP/1.1
+Content-Type: application/xml
+
+<host>
+    <name>$NAME</name>
+    <address>$ADDRESS</address>
+    <port>$PORT</port>>
+    <root_password>$PASSWORD</root_password>
+    <cluster>
+        <name>$CLUSTERNAME</name>
+    </cluster>
+    <dns_resolver_configuration>
+      <name_servers>
+        <name_server>1.1.1.1</name_server>
+        <name_server>2.2.2.2</name_server>
+      </name_servers>
+    </dns_resolver_configuration>
+</host>
+```
+
+#### Updating Management Network
+only management network can be updated with DNS configuration.
```

Comment:
It would be better to let any network have `dns_configuration`, because any network can become the management network of a future cluster. On each cluster, only the config of the management network applies.